tests/tracing_test.go

```diff
@@ -158,6 +161,137 @@ func TestTracing(t *testing.T) {
 	}
 }
 
+// This test is testing to ensure that correct number of navigation spans are created
+// and they are created in the correct order.
+func TestNavigationSpanCreation(t *testing.T) {
+	t.Parallel()
+
+	// Init tracing mocks
+	tracer := &mockTracer{
+		spans: make(map[string]struct{}),
+	}
+	tp := &mockTracerProvider{
+		tracer: tracer,
+	}
+	// Start test server
+	ts := httptest.NewServer(http.HandlerFunc(
+		func(w http.ResponseWriter, r *http.Request) {
+			fmt.Fprint(w, html)
+		},
+	))
+	defer ts.Close()
+
+	// Initialize VU and browser module
+	vu := k6test.NewVU(t, k6test.WithTracerProvider(tp))
+
+	rt := vu.Runtime()
+	root := browser.New()
+	mod := root.NewModuleInstance(vu)
+	jsMod, ok := mod.Exports().Default.(*browser.JSModule)
+	require.Truef(t, ok, "unexpected default mod export type %T", mod.Exports().Default)
+	require.NoError(t, rt.Set("browser", jsMod.Browser))
+	vu.ActivateVU()
+
+	testCases := []struct {
+		name     string
+		js       string
+		expected []string
+	}{
+		{
+			name: "goto",
+			js: fmt.Sprintf(`
+				page = await browser.newPage();
+				await page.goto('%s');
+				browser.closeContext();
+				`, ts.URL),
+			expected: []string{
+				"iteration",
+				"browser.newPage",
+				"browser.newContext",
+				"browserContext.newPage",
+				"navigation", // created when a new page is created
+				"page.goto",
+				"navigation", // created when a navigation occurs after goto
+			},
+		},
+		{
+			name: "reload",
```

Comment:
This test is failing in CI now. I need to try to fix it :/

Reply:
No idea why this happened. I need to keep an eye on it. Some CDP magic somewhere.